src/net.cpp

```diff
@@ -2300,11 +2300,14 @@ bool CConnman::InitBinds(const std::vector<CService>& binds, const std::vector<N
     }
     if (binds.empty() && whiteBinds.empty()) {
         struct in_addr inaddr_any;
-        inaddr_any.s_addr = INADDR_ANY;
+        inaddr_any.s_addr = htonl(INADDR_ANY);
         struct in6_addr inaddr6_any = IN6ADDR_ANY_INIT;
```

Comment:
nit. Why casting 0 to 0 , 0 is 0 in every byte-order? I like this to be a constant and not a hookable function.

Reply:
Is it guaranteed that `INADDR_ANY` will always be 0, on any operating system?

Reply:
[rfc1122](https://tools.ietf.org/html/rfc1122#section-3.2.1.3) section-3.2.1.3

Reply:
> Why casting 0 to 0 , 0 is 0 in every byte-order?

Firstly, https://www.man7.org/linux/man-pages/man7/ip.7.html:
> `in_addr` should be assigned one of the `INADDR_*` values using `htonl`...

Secondly, it is inconsistent to have `s_addr = INADDR_ANY` in a line, and `s_addr = htonl(INADDR_LOOPBACK)` in another one.

Reply:
Yup i guess its just a generation thing, its a nit, but i always liked to some degree how the bitcoin client did network things different than expected, 
U will never know who wrote the example codes and how drunk or biased by a gun on there head they where at those times when the foundations  u all take for granted where coded. 
INADDR_LOOPBACK is a different case and hey what can u do if the constant is not in net order, i remember the rant of Hobbit when he bashed on Berkeley sockets implementation ( also on many other reasons) . Htonl() was one of the first predictable ways the :Gesseler had to go". And if u bind to "this host" or maybe not makes a difference.

I will not way in this more, made my point and u are in charge its now ur world, but please never trade readability for function.

Reply:
@saibato Are you ok?

Reply:
`htonl()` is the proper one even though the number is a palindrome. We had some discussion about that [here](https://github.com/bitcoin/bitcoin/pull/19628#discussion_r464156291) and [here](https://github.com/bitcoin/bitcoin/pull/19628#discussion_r473486793).

Reply:
@sipa 
> @Saibato Are you ok?
<details><summary>click here :package: :tada:  . to find out!</summary>
<br>
I case u forgot,  just assign INADDR_ANY without htonl is up till now in source code right after some first commits. from November 2009 onward.
Even you later never changed that behavior and so far all run good with this.

I get nervous when things in bitcoin are changed by a drive by commit just because ...

Why even touch it?  Suppose Satoshi knew what we did in cold war and those where those magic lines that made bitcon a stable server.
I doubt that u lads are like we, but u all have the potential to grow and learn.

i like it when i can trigger u to mature, but ok is that all?  


```
    // The sockaddr_in structure specifies the address family,
    // IP address, and port for the socket that is being bound
    struct sockaddr_in sockaddr;
    memset(&sockaddr, 0, sizeof(sockaddr));
    sockaddr.sin_family = AF_INET;
    sockaddr.sin_addr.s_addr = INADDR_ANY; // bind to all IPs on this computer
    sockaddr.sin_port = DEFAULT_PORT;

```
```
commit 93cfb02acb13b74198b8e0ab72f039313ecf9f1f
Author:	s_nakamoto <s_nakamoto@1a98c847-1fd6-4fd8-948a-caf3550aa51b>  Fri Nov  6 05:50:05 2009
Committer:	s_nakamoto <s_nakamoto@1a98c847-1fd6-4fd8-948a-caf3550aa51b>  Fri Nov  6 05:50:05 2009
```
</details>